### PR TITLE
fix waitingforreceipt error

### PR DESCRIPTION
**Pull request by luu-alex**
## Description
#7007
This was an issue reported internally, waitforreceipt would fail when a new test geth env was set up,  
this will catch the error produced. Fixes this problem https://github.com/ethereum/go-ethereum/issues/28877 
Please include a summary of the changes and be sure to follow our [Contribution Guidelines](https://github.com/web3/web3.js/blob/4.x/.github/CONTRIBUTING.md).

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run lint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

**Review by avkos — APPROVED** [review]
LGTM. please increase test coverage

File: packages/web3-eth/src/rpc_method_wrappers.ts [hunk 1]
```diff
@@ -496,11 +496,16 @@ export async function getTransactionReceipt<ReturnFormat extends DataFormat>(
...
-	);
-
+	let response;
+	try {
+		 response = await ethRpcMethods.getTransactionReceipt(
+			web3Context.requestManager,
+			transactionHashFormatted,
+		);
+	} catch (error) {
+		console.warn(error)
```
Comment (jdevcs) [inline packages/web3-eth/src/rpc_method_wrappers.ts]: this will only warn instead of throw in other cases?, and thats not recommended

Comment (luu-alex) [reply packages/web3-eth/src/rpc_method_wrappers.ts]: Updated, thanks

File: packages/web3-eth/src/rpc_method_wrappers.ts [hunk 2]
```diff
@@ -496,11 +496,22 @@ export async function getTransactionReceipt<ReturnFormat extends DataFormat>(
...
+		 response = await ethRpcMethods.getTransactionReceipt(
+			web3Context.requestManager,
+			transactionHashFormatted,
+		);
+	} catch (error) {
+		// geth indexing error, we poll until transactions stopped indexing
+		if (typeof error === 'object' && !isNullish(error) && 'message' in error && (error as { message: string }).message === 'transaction indexing is in progress') { 
+			console.warn('Transaction indexing is in progress.')
+		} else {
+			throw error;
```
Comment (jdevcs) [inline packages/web3-eth/src/rpc_method_wrappers.ts:510]: This also should be tested locally with Geth before merging.

Comment (luu-alex) [reply packages/web3-eth/src/rpc_method_wrappers.ts:510]: Yup tested locally and works well

**Pull request merged**